### PR TITLE
ZEPPELIN-181 Provide more Feedback on UI when Clicking the Cancel/Stop Button of a Pending/Running Paragraph

### PR DESCRIPTION
**Pull request by r-kamath**
On branch ZEPPELIN-181
 Changes to be committed:
	modified:   zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js
	modified:   zeppelin-web/src/app/notebook/paragraph/paragraph.html

**screenshot** :

### CODE REVIEW

**Comment by prabhjyotsingh** [comment]
Tried on my local, LGTM.

File: zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js [hunk 1]
```diff
@@ -245,6 +245,7 @@ angular.module('zeppelinWebApp')
   $scope.cancelParagraph = function() {
     console.log('Cancel %o', $scope.paragraph.id);
     websocketMsgSrv.cancelParagraphRun($scope.paragraph.id);
+    $scope.paragraph.status = 'CANCELING';
```
Comment (felixcheung) [inline zeppelin-web/src/app/notebook/paragraph/paragraph.controller.js]: perhaps independent from this visual change, not everything is cancel-able from the interpreter? should there be a way to communicate that to the visual?

**Comment by corneadoug** [comment]
I think this should be handled at the Websocket/Backend level.

For example, after sending the CANCEL_PARAGRAPH, you will receive the note with paragraph.status as 'ABORT' once it is finished.

While you did set the pagraph.status to 'CANCELING', other people visiting the note would see it as 'PENDING' instead of 'CANCELING'

**Comment by r-kamath** [comment]
@corneadoug @felixcheung thanks for the feedback. trying to make some changes in the server side to have CANCELING status. will post my updates here

**Comment by r-kamath** [comment]
@corneadoug let me know if the update is good. Currently the new status `CANCELING` is used only for `cancelParagraph` server call and in the UI.

@felixcheung @Leemoonsoo 
>perhaps independent from this visual change, not everything is cancel-able from the interpreter? 

For a complete implementation this status should be used in Scheduler classes as well

**Comment by Leemoonsoo** [comment]
It'll broadcast 'CANCELING' event to connected browsers that is seeing the notebook.
However, if browser refreshes, it'll not see 'CANCELING' status any more.

As you mentioned, for completeness, i think it's better insert `CANCELING` status into state transition in Scheduler class.

**Comment by djoelz** [comment]
and don't forget your unit tests :)

**Comment by r-kamath** [comment]
will reopen when the patch is ready

**Pull request closed without merging**